test/unit/BPool/BPool_SwapExactAmountIn.tree

```diff
@@ -0,0 +1,26 @@
+BPool::SwapExactAmountIn
+├── when reentrancy lock is set
+│   └── it should revert
+├── when pool is not finalized
+│   └── it should revert
+├── when token in is not bound
+│   └── it should revert
+├── when token out is not bound
+│   └── it should revert
+├── when token in exceeds max allowed ratio
+│   └── it should revert
+├── when spot price before swap exceeds maxPrice
+│   └── it should revert
+├── when spot price after swap exceeds maxPrice
+│   └── it should revert
```

Comment:
isn't there an intermediary step? if spot price before exceeds spot price after? 🤔 

i do think this error is unreachable, but perhaps with smock we can have a chance of getting to it